operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -105,6 +105,12 @@ protected Integer currentScale(String namespace, String name) {
         return waitFor(reconciliation, namespace, name, "observed", pollIntervalMs, timeoutMs, this::isObserved);
     }
 
+    public Future<Void> deploymentReadiness(Reconciliation reconciliation, String namespace, String name, long pollIntervalMs, long timeoutMs) {
```

Comment:
Why don't you overwide the `readiness(...)` method here?

Reply:
That way i would also be covered by whatever tests there already are.

Reply:
This method calls both waitforobserve and readiness so, would it be okay to remove the part of waitforObserve from the tests
?

Reply:
I'm not sure I follow the question. The original idea was that the `waitForObserved` should be called from the regular readiness methods.

Reply:
Earlier `waitForObserve` method and `readiness` both of them were different right, but now since Iam calling `waitforObserve` from inside of `readiness` method. Its working fine for the Assembly Operator classes but I was asking that we are also using mocking in the test for both `readiness` and `waitforObserve` so if we call readiness, It also calls `waitforObserve` and again we also have a seperate mock for `waitforObserve`. for eg this--> ```        when(mockDepOps.waitForObserved(any(), anyString(), anyString(), anyLong(), anyLong())).thenReturn(
                Future.succeededFuture()
        );
        when(mockDepOps.readiness(any(), anyString(), anyString(), anyLong(), anyLong())).thenReturn(
                Future.succeededFuture()
        );```

Reply:
So here readiness will make a call to `waitForObserve` so is it okay to keep the above mocked `waitforObserve`?

Reply:
But you didn't added it to `readiness` ... you created a brand new method `deploymentReadiness`.

Reply:
Is mocked `waitForObserve` still needed? If yes, then it is ok to keep it. If it si not needed anymore, then don't keep it.

Reply:
> But you didn't added it to `readiness` ... you created a brand new method `deploymentReadiness`.

Actually I overrided `readiness` earlier but was a bit cinfused and didnt wanted to touch the test part much, that why I created a new method

Reply:
I will override the readiness method now then see if the test are passing perfectly or not

Reply:
@scholzj Overrriding the readiness method will also cause some problems since it is meddling with the DeploymenOperatorTest.

Reply:
I guess the error comes because of the `waitforObserve` call.

Reply:
Then you might need to fix the text 😉  ... you can override it to add more mocks if needed etc.